### PR TITLE
Try to migrate to Python 3.10 and all the newest to avoid bugs alread…

### PR DESCRIPTION
**Pull request by silverv**
…y fixed upstream

This is work in progress. Made a mess with submodules at first.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,12 +1,6 @@
...
-[submodule "kucher/libraries/construct"]
-	path = kucher/libraries/construct
-	url = https://github.com/construct/construct
 [submodule "kucher/libraries/dataclasses"]
 	path = kucher/libraries/dataclasses
-	url = https://github.com/ericvsmith/dataclasses
-[submodule "kucher/libraries/quamash"]
-	path = kucher/libraries/quamash
-	url = https://github.com/harvimt/quamash
+	url = https://github.com/ericvsmith/dataclasses
```
Comment (pavel-kirienko) [inline .gitmodules:6]: This should no longer be needed because in Python 3.7 onwards this is part of the standard library.

**Review by pavel-kirienko — CHANGES_REQUESTED** [review]
You seem to have added construct and quamash as pip dependencies instead of git submodules, which is wise. But then you added the entire source code of these libraries into this project's tree, what is that for? Was that an accident?

File: .gitmodules (hunk 1 above)
Comment (silverv) [reply .gitmodules:6]: It is an accident, but I want to fork them and have our own copies

Comment (pavel-kirienko) [reply .gitmodules:6]: You can do it but what for? These are relatively popular libraries and there should be little risk of them disappearing overnight (assuming you are talking about construct and quamash). Even if we did fork them, surely we don't want to bother with maintaining independent packages on PyPI.

**Comment by silverv** [comment]
I would close this as it was an attempt that was superseded by the latest successful attempt #48

**Pull request closed without merging**